### PR TITLE
refactor: additionalProperties interpretation

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds the refactoring of the interpretation for `additionalProperties`.

It changes the entire perspective, to instead of adding simplification functions per CommonModel property, we process the JSON Schema keywords which then affects the CommonModel in some way.

**It adds**
- InterpretAdditionalProperties, to interpret the `additionalProperties` schema keyword.

**Files this PR is supposed to replace**
- https://github.com/asyncapi/generator-model-sdk/blob/master/src/simplification/SimplifyAdditionalProperties.ts
- And all related tests have been refactored to have next to no dependencies by using the jest mocking framework.

**Todo**

- [x] Refactor docs
- [x] Solve code smells
- [x] Add logging to the process
- [x] Inspect tests
- [x] Rename to transformer

**Related issue(s)**
Blocked by #184

### CODE REVIEW

File: src/interpreter/InterpretAdditionalProperties.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Interpreter function for JSON Schema draft 7 additionalProperties keyword.
+ * 
+ * @param schema
+ * @param model
+ * @param interpreter
+ */
+export default function interpretAdditionalProperties(schema: Schema | boolean, model: CommonModel, interpreter : Interpreter) {
+  if (typeof schema === 'boolean' || !isModelObject(model)) return;
+  const newModels = interpreter.interpret(schema.additionalProperties || true);
```
Comment (magicmatatjahu) [inline src/interpreter/InterpretAdditionalProperties.ts:15]: If you have `true` then you never go to `15` line so

```ts
const newModels = interpreter.interpret(schema.additionalProperties);
```

is enough.

Comment (jonaslagoni) [reply src/interpreter/InterpretAdditionalProperties.ts:15]: There is a difference between the schema being boolean and the additionalProperty schema being a boolean.

Not sure what you are suggesting here? 🤔

Comment (jonaslagoni) [reply src/interpreter/InterpretAdditionalProperties.ts:15]: waiiit, `schema` should not have type `boolean | schema`... 

Gonna change it

**Review by magicmatatjahu — APPROVED** [review]

**Comment by jonaslagoni** [comment]
Thanks!

**Pull request merged**